libs/datasets/nytimes_anomalies.py

```diff
@@ -0,0 +1,179 @@
+
+
+NYTIMES_ANOMALIES_CSV = dataset_utils.LOCAL_PUBLIC_DATA_PATH / pathlib.Path(
+    "data/cases-nytimes/anomalies.csv"
+)
+
+
+@enum.unique
+class NYTimesFields(GetByValueMixin, ValueAsStrMixin, FieldName, enum.Enum):
+    """Fields used in the NYTimes anomalies file"""
+
+    DATE = "date"
+    END_DATE = "end_date"
+    COUNTY = "county"
+    STATE = "state"
+    GEOID = "geoid"
+    TYPE = "type"
+    OMIT_FROM_ROLLING_AVERAGE = "omit_from_rolling_average"
+    OMIT_FROM_ROLLING_AVERAGE_ON_SUBGEOGRAPHIES = "omit_from_rolling_average_on_subgeographies"
+    DESCRIPTION = "description"
+
+
+@lru_cache(None)
+def read_nytimes_anomalies():
+    df = pd.read_csv(
+        NYTIMES_ANOMALIES_CSV, parse_dates=[NYTimesFields.DATE, NYTimesFields.END_DATE]
+    )
+
+    # Extract fips from geoid column.
+    df[CommonFields.FIPS] = df[NYTimesFields.GEOID].str.replace("USA-", "")
+
+    # Denormalize data so that each row represents a single date+location+metric anomaly
+    df = _denormalize_nyt_anomalies(df)
+
+    # Add LOCATION_ID column (must happen after denormalizing since denormalizing can add additional
+    # rows for subgeographies).
+    df[CommonFields.LOCATION_ID] = df[CommonFields.FIPS].map(dataset_utils.get_fips_to_location())
+
+    # A few locations (e.g. NYC aggregated FIPS 36998) don't have location IDs. That's okay, just remove them.
+    df = df.loc[df[CommonFields.LOCATION_ID].notna()]
+
+    # Convert "type" column into "variable" column using new_cases / new_deaths as the variable.
+    assert df[NYTimesFields.TYPE].isin(["cases", "deaths"]).all()
+    df[PdFields.VARIABLE] = df[NYTimesFields.TYPE].map(
+        {"cases": CommonFields.NEW_CASES, "deaths": CommonFields.NEW_DEATHS}
+    )
+
+    # Add demographic bucket (all) to make it more compatible with our dataset structure.
+    df[PdFields.DEMOGRAPHIC_BUCKET] = "all"
+
+    return df
+
+
+# TODO(mikelehen): This should probably live somewhere more central, but I'm not sure where.
+def _get_county_fips_codes_for_state(state_fips_code: str) -> List[str]:
+    """Helper to get county FIPS codes for all counties in a given state."""
+    geo_data = dataset_utils.get_geo_data()
+    state = geo_data.set_index("fips").at[state_fips_code, "state"]
+    counties_df = geo_data.loc[
+        (geo_data["state"] == state) & (geo_data["aggregate_level"] == "county")
+    ]
+    counties_fips = counties_df["fips"].to_list()
+    return counties_fips
+
+
+def _denormalize_nyt_anomalies(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    The NYT anomaly data is normalized such that each row can represent an
+    anomaly for multiple dates, locations, and metrics. We want to denormalize
+    it so that each row represents a single date+location+metric anomaly.
+    """
+
+    # Look for rows with an end_date and create separate rows for each date in the [date, end_date] range.
+    df = df.assign(
+        date=df.apply(
+            lambda row: pd.date_range(
```

Comment:
Might have a preference for making this a function - generally try to avoid multi line lambdas in any form

Reply:
Sure, done.